### PR TITLE
Vb 2110 update existing sessions to use the new enhanced type method

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Updates the incentive logic to use grouping instead of a boolean.
## What is the intent behind these changes?

To bring the sessions for incentive levels logic in line with category and housing locations.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt:56]: dont we need this?

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt:56]: Updated to include incentiveLevelGroupReferences.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:118]: where has this gone?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/PrisonerCategoryMatcher.kt
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/PrisonerCategoryMatcher.kt:11]: nice

File: src/main/resources/session-template-data/template.ftl
Comment (AledWEvans) [inline src/main/resources/session-template-data/template.ftl:219]: awesome

**Review by AledWEvans — APPROVED** [review]
LGTM, but wonder if you should do VB-2216 before you merge this in!?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:118]: moved this into PrisonerSessionValidator

**Pull request merged**
